pubtools/_pulp/tasks/garbage_collect.py

```diff
@@ -96,13 +99,32 @@ def clean_all_rpm_content(self):
             LOG.info("No repos found for cleaning.")
             return
         arc_repo = clean_repos[0]
-        deleted_arc = list(arc_repo.remove_content(criteria=criteria))
+
         deleted_content = []
-        for task in deleted_arc:
-            if task.repo_id == "all-rpm-content":
+
+        while True:
+            delete_arc_tasks = list(
+                arc_repo.remove_content(
+                    criteria=criteria, limit=UNASSOCIATE_BATCH_LIMIT
+                )
+            )
+            # The pulp client returns task(s) wrapped in a list
+            arc_tasks = list(
+                filter(lambda x: x.repo_id == "all-rpm-content", delete_arc_tasks)
+            )
+            for task in arc_tasks:
                 for unit in task.units:
                     LOG.info("Old all-rpm-content deleted: %s", unit.name)
                     deleted_content.append(unit)
+
+            # Assuming the client can return multiple (python) tasks.
+            # If all tasks delete less units than the batch limit,
+            # implying all relevant units have been unassociated.
+            if not arc_tasks or not any(
```

Comment:
Not sure about the second half of this condition.

In pseudocode this all seems to mean:

```
if
    not (there were tasks)
    or
    not (there was a task which unassociated less than the limit)
then stop looping
```

Should that second `not` really be there?  If there *was not* a task which unassociated less than the limit, isn't that an indication that we need to keep going as there are likely still units to be processed?